src/System.Private.CoreLib/shared/System/Reflection/MemberInfo.cs

```diff
@@ -46,28 +47,48 @@ public virtual Module Module
 
         public static bool operator ==(MemberInfo left, MemberInfo right)
         {
-            if (object.ReferenceEquals(left, right))
+            if (ReferenceEquals(left, right))
+            {
                 return true;
+            }
 
-            if ((object)left == null || (object)right == null)
+            if (left is null || right is null)
+            {
                 return false;
+            }
+
+            if (left is Type type1)
```

Comment:
Should this rather call `left.Equals(right)` instead of this manual dispatch?

Reply:
Looks like `Type` is the only particularly special one where `operator ==` is an `external` call and its `.Equals` test is a `ReferenceEquals`; so maybe have to manually dispatch for `Type`?

Reply:
`operator==` is external call on Type just so the JIT can recognize it as intrinsic. It is from the days when the method had to be FCall in order to be recognized as intrinsic.

It should be fine to just call Equals for Type as well.